### PR TITLE
Support for scale property on model overlays

### PR DESCRIPTION
**Pull request by hyperlogic**
If "scale" property is present but "dimensions" are not, the model's natural dimensions will be scaled by the scale property.

If only the "dimensions" property is present, the model will be automatically scaled to fit the dimensions specified.

If both properties are present, the model will still be "scale to fit" but the dimension will be scaled by the scale factor.

For example:

If a model is loaded that is 2cm tall, is loaded with no "dimensions" or "scale" properties.
It will be displayed as 2cm tall.

    {"scale": 2}

The above properties will result in a model that is 4cm tall.

    {"dimensions": 1}

This will result in a model that is 1cm tall.

    {"scale": 2, "dimensions": 2}

Will result in a model that is 2cm tall.

### CODE REVIEW

**Comment by hyperlogic** [comment]
https://highfidelity.fogbugz.com/f/cases/1144/Fix-issue-with-large-fish-in-home-content-set

**Comment by hifi-gustavo** [comment]
Build console output for commit: 1d77cec125876844b3c4afced504aa475db9a094 available at: https://gist.github.com/dc8c79dea19ddcd6d866bbc90621688f<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8238-1d77cec.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8238-1d77cec.dmg

File: interface/src/ui/overlays/ModelOverlay.cpp [hunk 1]
```diff
@@ -84,32 +87,49 @@ void ModelOverlay::render(RenderArgs* args) {
...
 
-    _updateModel = true;
+    auto dimensions = properties["dimensions"];
+    if (dimensions.isValid()) {
+        _scaleToFit = true;
+        setDimensions(vec3FromVariant(dimensions));
+    } else if (scale.isValid()) {
+        // if "scale" property is set but "dimentions" is not.
+        // do NOT scale to fit.
+        if (scale.isValid()) {
```
Comment (huffman) [inline interface/src/ui/overlays/ModelOverlay.cpp]: It looks like you check `scale.isValid()` twice, once here and once in the `else if`

**Comment by hyperlogic** [comment]
Test plan:
* When connecting to the home sandbox content on localhost, the fish should be regular size. (on master they are very large). 
* Open the running scripts dialog.
* run the following [script url](https://gist.githubusercontent.com/hyperlogic/a2e63ac0e70d2e9d132bcc37316c6f30/raw/c3219a88d3f7bd51f58c88261ff14ff486c1c54a/modelOverlayScaleTest.js)
* You should see the following four models appear from left to right:
   * The axes.fbx model that is one meter by one meter in size.
   * The axes.fbx model scaled to 1/2 meter in size.
   * The axes.fbx model scaled to 1/2 meter in size.
   * The axes.fbx model scaled to 1/2 meter in size.

File: interface/src/ui/overlays/ModelOverlay.cpp (hunk 1 above)
Comment (hyperlogic) [inline interface/src/ui/overlays/ModelOverlay.cpp]: whoops, I submitted a fix.

**Comment by hifi-gustavo** [comment]
Build console output for commit: 789297d8496cbb9228d12123ffc691f3ae70ab02 available at: https://gist.github.com/dc8c79dea19ddcd6d866bbc90621688f<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8238-789297d.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8238-789297d.dmg

**Comment by Nex-Pro** [comment]
**PASS**

**Tested on:** OSX El Capitan.

Overlays render correctly.

Fish render at correct size.

***Rightmost overlay is poking through the wall but is fine.***

screen shot 2016-07-14 at 12 29 37
screen shot 2016-07-14 at 12 30 02

**Pull request merged**